### PR TITLE
Mute E_DEPRECATED warnings during tests and add depreciation warnings

### PR DESCRIPTION
**Pull request by cleptric**
First batch of added deprecation warnings. How atomic should we do this?
Also, which test should be updated?
For example `EntityContextTest::testValSchemaDefault` calls `$table->schema()->column('title');`.
Should this become `getColumn()`?
We also have a whole test file for `Cake\Database\Schema\Table`

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <php>
         <ini name="memory_limit" value="-1"/>
         <ini name="apc.enable_cli" value="1"/>
+        <ini name="error_reporting" value="E_ALL ~ &amp;E_USER_DEPRECATED"/>
```
Comment (markstory) [inline phpunit.xml.dist]: I'm not sure this will work. I think you might need to use `16383` with a comment on where that value came from.

**Review by markstory — COMMENTED** [review]
Thanks for getting this started!

**Comment by markstory** [comment]
@cleptric I fixed up the constant value and fixed the deprecation warnings I could easily find.

**Comment by antograssiot** [comment]
I also removed deprecations and add a specific error_reporting call has it doesn't seems to works bu using PHPUnit php ini section

**Comment by cleptric** [comment]
Thank you @markstory and @antograssiot 
So I guess the way to go is to update all deprecated method calls to use the new methods.

**Comment by markstory** [comment]
>  So I guess the way to go is to update all deprecated method calls to use the new methods.

I think updating our internal usage of deprecated methods will make the transition to 4.x easier for us in the future.

**Comment by cleptric** [comment]
Besides the phpstan issues (not related to the changes here) I think we good now 🙂

**Pull request merged**